HPC/HPC_Embeddings/openvino_env/run_embeddings.sh

```diff
@@ -0,0 +1,211 @@
+#!/bin/bash
+# Script to run embeddings generation for acts or regulations
+# Usage: ./run_embeddings.sh acts|regs [options]
+
+# Display help function
+show_help() {
+    echo "Usage: $0 acts|regs [options]"
+    echo ""
+    echo "Required arguments:"
+    echo "  acts|regs                      - Collection type to process"
+    echo ""
+    echo "Optional arguments:"
+    echo "  --help                         - Show this help message and exit"
+    echo "  --folder-range RANGE           - Folder range to process (e.g., '40-43' or 'all') [default: all]"
+    echo "  --batch-size SIZE              - Batch size for processing [default: 16]"
+    echo "  --num-streams NUM              - Number of streams [default: 8]"
+    echo "  --download-model yes|no        - Download OpenVINO model from HuggingFace [default: no]"
+    echo "  --base-dir DIR                 - Custom base directory for input files"
+    echo "  --output-dir DIR               - Custom output directory for embeddings"
+    echo ""
+    echo "Default directories (if not specified):"
+    echo "  For acts:"
+    echo "    Base dir: /opt/app-root/src/Ashivaku/citz-imb-ai/HPC/HPC_Embeddings/build/Processed/acts"
+    echo "    Output dir: Embeddings/acts"
+    echo "  For regs:"
+    echo "    Base dir: /opt/app-root/src/Ashivaku/citz-imb-ai/HPC/HPC_Embeddings/BCLaws_Output/reg_flat"
+    echo "    Output dir: Embeddings/regs"
+    echo ""
+    echo "Examples:"
+    echo "  $0 acts                        - Process all acts with default settings"
+    echo "  $0 regs --folder-range 40-43   - Process regulations in folders 40-43"
+    echo "  $0 acts --download-model yes   - Process acts and download the model"
+    echo "  $0 acts --base-dir /path/to/acts --output-dir /path/to/output --batch-size 32"
+    echo "                                 - Process acts using custom directories and batch size"
+    exit 0
+}
+
+# Check if help is requested
+if [ "$#" -eq 0 ] || [ "$1" == "--help" ] || [ "$2" == "--help" ]; then
+    show_help
+fi
```

Comment:
Something I learned today, you could use `"$@" == *"--help"*` to detect any argument that has `--help`.